pkg/kubelet/nodestatus/setters.go

```diff
@@ -123,11 +123,19 @@ func NodeAddress(nodeIPs []net.IP, // typically Kubelet.nodeIPs
 		if externalCloudProvider {
 			// If --cloud-provider=external and node address is already set,
 			// then we return early because provider set addresses should take precedence.
-			// Otherwise, we try to look up the node IP and let the cloud provider override it later
+			// Otherwise, we try to use the node IP defined via flags and let the cloud provider override it later
 			// This should alleviate a lot of the bootstrapping issues with out-of-tree providers
 			if len(node.Status.Addresses) > 0 {
 				return nil
 			}
+			// If nodeIPs are not specified wait for the external cloud-provider to set the node addresses.
+			// Otherwise uses them on the assumption that the installer/administrator has the previous knowledge
+			// required to ensure the external cloud provider will use the same addresses to avoid the issues explained
+			// in https://github.com/kubernetes/kubernetes/issues/120720.
+			// We are already hinting the external cloud provider via the annotation AnnotationAlphaProvidedIPAddr.
+			if !nodeIPSpecified {
+				return nil
+			}
```

Comment:
Maybe just squash these two into a single `if` / comment?

Reply:
I thought that but in terms of code readability I found this split easier to understand and the git blame will keep the history for both blocks ... in terms of performance is the same , I expect the compiler to do the same thing

Reply:
not a huge deal either way, but personally, I appreciate the second if block with the dedicated comment and the preserved history